alpha version

odinuv opened the pull request.

odinuv committed:
feat: first version

odinuv committed:
fix: no integration test yet
chore: cs fixes

odinuv committed:
fix: syntax and enable integration test

odinuv committed:
chore: no functional tests yet

odinuv committed:
Merge branch 'master' of https://github.com/keboola/wr-storage into odin-first-version

odinuv committed:
chore: enable integration test

odinuv committed:
docs: update instructions

odinuv committed:
feat: write to existing table too

odinuv committed:
fix: parallel tests

odinuv committed:
chore: remove unused import

odinuv committed:
fix: cannot be empty

Halama commented on tests/phpunit/StorageWriterTest.php:
```diff
@@ -0,0 +1,119 @@
+            'token' => getenv('KBC_TEST_TOKEN'),
+            'url' => getenv('KBC_TEST_URL'),
+        ]);
+        $tables = $this->client->listTables('in.c-wr-storage-test');
```
testy očekávájí že ten bucket `in.c-wr-storage-test` existuje, mělo by to být minimálně zmíněno v README

Halama replied:
možná by taky bylo lepší když by ten bucket byl v ENV `KBC_TEST_BUCKET`

Halama commented on src/Component.php:
```diff
@@ -2,15 +2,53 @@
+        $config = $this->getConfig();
+        $client = new Client(['token' => $config->getToken(), 'url' => $config->getUrl()]);
+        $tokenInfo = $client->verifyToken();
+        if (!$tokenInfo['bucketPermissions']) {
```
v `bucketPermissions` se vracejí vždy všechny buckety na které má token práva. Pokud je to prázdné znamená to že nemá právo na žádný bucket. Myslím že tahle podmínka je tady zbytečná.

Halama commented on src/Component.php:
```diff
@@ -2,15 +2,53 @@
+            throw new UserException("The token has too broad permissions.");
+        }
+        $bucket = $config->getBucket();
+        if (empty($tokenInfo['bucketPermissions'][$bucket]) || ($tokenInfo['bucketPermissions'][$bucket] != 'write')) {
```
Zároveň zkontrolovat že to je jediný bucket do kterého může zapisovat? A pokud tomu tak není skončit chybou?

Halama commented on README.md:
````diff
@@ -1,12 +1,32 @@
+		"url": "https://connection.keboola.com/"
+	}
+}
+```
````
popsat jaké `ENV` to potřebuje mít nastavené v `.env`

Halama commented on README.md:
```diff
@@ -1,12 +1,32 @@
 [![Build Status](https://travis-ci.org/keboola/my-component.svg?branch=master)](https://travis-ci.org/keboola/my-component)
 
-> Fill in description
+Writes tables from input mapping to the configured bucket in the destination project.
```
Popsat jak připravit bucket a token? dokud nebude ui.

Halama replied:
koukám že je tam teda configuration schema, ale pro vývoj by to bylo fajn mít v readme.

Halama committed:
dev - tests env variables

Halama commented on src/Component.php:
```diff
@@ -2,15 +2,53 @@
+            try {
+                $tableId = $bucket . '.' . $table['destination'];
+                if ($client->tableExists($tableId)) {
+                    $client->deleteTableRows($tableId);
```
zbytečný když hned potom děláš full load.

Halama commented on tests/phpunit/StorageWriterTest.php:
```diff
@@ -0,0 +1,119 @@
+        }
+    }
+
+    public function tearDown() : void
```
zbytečný:)

Halama reviewed (changes requested):
Hodil jsem ti pár komentářů, bral jsem to spíš z funkčního hlediska. Dovolil jsem si tam poslat jeden commit aby mi jely jednoduše testy.

odinuv committed:
fix: token permissions check
fix: unnecessary row delete
docs: clarify README
tests: KBC_TEST_BUCKET is now used instead of hardcoded bucket name

odinuv committed:
Merge
